deploy/runtools/firesim_topology_elements.py

```diff
@@ -442,68 +507,64 @@ def supernode_get_num_siblings_plus_one(self):
+                return node
+        assert False, "Should return supernode sibling"
 
-    def supernode_get_sibling_mac_address(self, siblingindex):
-        """ return the sibling's mac address for supernode mode.
-        siblingindex = 1 -> next sibling, 2 = second, 3 = last one."""
-        return self.supernode_get_sibling(siblingindex).get_mac_address()
-
-    def supernode_get_sibling_rootfs(self, siblingindex):
-        """ return the sibling's rootfs for supernode mode.
-        siblingindex = 1 -> next sibling, 2 = second, 3 = last one."""
-        return self.supernode_get_sibling(siblingindex).get_rootfs_name()
-
-    def supernode_get_sibling_bootbin(self, siblingindex):
-        """ return the sibling's rootfs for supernode mode.
-        siblingindex = 1 -> next sibling, 2 = second, 3 = last one."""
-        return self.supernode_get_sibling(siblingindex).get_bootbin_name()
-
-    def supernode_get_sibling_rootfs_path(self, siblingindex):
-        return self.supernode_get_sibling(siblingindex).get_job().rootfs_path()
-
-    def supernode_get_sibling_bootbinary_path(self, siblingindex):
-        return self.supernode_get_sibling(siblingindex).get_job().bootbinary_path()
-
-    def supernode_get_sibling_link_latency(self, siblingindex):
-        return self.supernode_get_sibling(siblingindex).server_link_latency
-
-    def supernode_get_sibling_bw_max(self, siblingindex):
-        return self.supernode_get_sibling(siblingindex).server_bw_max
-
-    def supernode_get_sibling_shmemportname(self, siblingindex):
-        return self.supernode_get_sibling(siblingindex).uplinks[0].get_global_link_id()
-
-    def run_sim_start_command(self, slotno):
+    def run_sim_start_command(self, slotno: int) -> None:
         """ get/run the command to run a simulation. assumes it will be
         called in a directory where its required_files are already located."""
 
         num_siblings = self.supernode_get_num_siblings_plus_one()
 
-        all_macs = [self.get_mac_address()] + [self.supernode_get_sibling_mac_address(x) for x in range(1, num_siblings)]
-        all_rootfses = self.process_qcow2_rootfses([self.get_rootfs_name()] + [self.supernode_get_sibling_rootfs(x) for x in range(1, num_siblings)])
-        all_bootbins = [self.get_bootbin_name()] + [self.supernode_get_sibling_bootbin(x) for x in range(1, num_siblings)]
-        all_linklatencies = [self.server_link_latency] + [self.supernode_get_sibling_link_latency(x) for x in range(1, num_siblings)]
-        all_maxbws = [self.server_bw_max] + [self.supernode_get_sibling_bw_max(x) for x in range(1, num_siblings)]
+        all_macs = [self.get_mac_address()]
+        all_rootfses = [self.get_rootfs_name()]
+        all_bootbins = [self.get_bootbin_name()]
+        all_linklatencies = [self.server_link_latency]
+        all_maxbws = [self.server_bw_max]
+
+        for x in range(1, num_siblings):
+            sibling = self.supernode_get_sibling(x)
```

Comment:
Can probably convert this back into the `[self.supernode_get_sibling(x).FOO_TO_CALL]` syntax